subprojects/logging/src/test/groovy/org/gradle/internal/logging/sink/ProgressLogEventGeneratorTest.groovy

```diff
@@ -17,468 +17,145 @@
+        then: 1 * downstreamListener.onOutput({ it.toString() == "[WARN] [category] Warning: some deprecation or something" })
+        then: 1 * downstreamListener.onOutput(endBuildEvent)
+        then: 0 * _
     }
 
-    def handlesMultipleOperationsInProgressAtOnce() {
-        ProgressLogEventGenerator generator = new ProgressLogEventGenerator(target, true)
+    def "handles multiple simultaneous operations"() {
         given:
-        def firstStart = startWithHeader("task1")
-        def secondStart = startWithHeader("task2")
-        def secondProgress = progress("task2-progress")
-        def secondComplete = complete("task2-done")
-        def firstComplete = complete("task1-done")
-        when:
-        generator.onOutput(firstStart)
-        generator.onOutput(secondStart)
-        generator.onOutput(firstComplete)
-        generator.onOutput(secondProgress)
-        generator.onOutput(secondComplete)
-
-        then:
-        1 * target.onOutput({ StyledTextOutputEvent event ->
-            event.spans.size() == 1 && event.spans[0].text == toNative('task2\n')
-        })
-        1 * target.onOutput({ StyledTextOutputEvent event ->
-            event.spans.size() == 3 &&
-                event.spans[0].text == toNative('task1 ') &&
-                event.spans[1].text == toNative('task1-done') &&
-                event.spans[2].text == toNative('\n')
-        })
-        1 * target.onOutput({ StyledTextOutputEvent event ->
-            event.spans.size() == 3 &&
-                event.spans[0].text == toNative('task2 ') &&
-                event.spans[1].text == toNative('task2-done') &&
-                event.spans[2].text == toNative('\n')
-        })
-        0 * target._
+        ProgressLogEventGenerator listener = new ProgressLogEventGenerator(downstreamListener, true, timeProvider)
+        def taskAStartEvent = new ProgressStartEvent(new OperationIdentifier(-3L), new OperationIdentifier(-4L), tenAm, CATEGORY, "Execute :a", ":a", null, null, new OperationIdentifier(2L), null, BuildOperationType.TASK)
+        def taskBStartEvent = new ProgressStartEvent(new OperationIdentifier(-5L), new OperationIdentifier(-6L), tenAm, CATEGORY, "Execute :b", ":b", null, null, new OperationIdentifier(3L), null, BuildOperationType.TASK)
+        def taskAOutput = event('message for task a', LogLevel.WARN, taskAStartEvent.buildOperationId)
+        def taskBOutput = event('message for task b', LogLevel.WARN, taskBStartEvent.buildOperationId)
+        def taskBCompleteEvent = new ProgressCompleteEvent(taskBStartEvent.progressOperationId, tenAm, CATEGORY, 'Complete: :b', null)
+        def taskACompleteEvent = new ProgressCompleteEvent(taskAStartEvent.progressOperationId, tenAm, CATEGORY, 'Complete: :a', 'UP-TO-DATE')
+
+        when: listener.onOutput([taskAStartEvent, taskBStartEvent, taskAOutput, taskBOutput, taskBCompleteEvent, taskACompleteEvent])
+
+        then: 1 * downstreamListener.onOutput({ it.toString() == "[QUIET] [category] " })
```

Comment:
My understanding is that this is necessary to assert order of calls, but I'll double-check. Order matters because it affects display.